module/usb_disk_mode.c

```diff
@@ -25,17 +24,31 @@
 #include "uhi_msc_mem.h"
 #include "usb_protocol_msc.h"
 
+// Local functions for usb filesystem serialization
+void tele_usb_putc(void* self_data, uint8_t c);
+void tele_usb_write_buf(void* self_data, uint8_t* buffer, uint16_t size);
+uint16_t tele_usb_getc(void* self_data);
+bool tele_usb_eof(void* self_data);
 
-static uint8_t grid_state = 0;
-static uint16_t grid_count = 0;
-static uint8_t grid_num = 0;
+void tele_usb_putc(void* self_data, uint8_t c) {
+    file_putc(c);
+}
 
-static void grid_usb_write(scene_state_t *scene);
-static void grid_usb_read(scene_state_t *scene, char c);
+void tele_usb_write_buf(void* self_data, uint8_t* buffer, uint16_t size) {
+    file_write_buf(buffer, size);
+}
 
+uint16_t tele_usb_getc(void* self_data) {
+    return file_getc();
+}
+
+bool tele_usb_eof(void* self_data) {
+    return file_eof() != 0;
+}
 
+// usb disk mode entry point
 void tele_usb_disk() {
-    char input_buffer[32];
+    char text_buffer[40];
```

Comment:
this is the fix to the `strcat` overflow. I noticed it because I initially put some additional local variables before this line, and they got corrupted by the end of the write cycle. I'm not sure what the consequences of this overflow would be, depends on the AVR32 stack frame layout, but it could theoretically mangle the return pointer, or stackvars in the calling function (like the saved irq flags?)